File: RootOfQuadraticEquation.cs
// mahmudnotes.com

using System;
namespace RootOfQuadraticEquation
{
    class Program
    {
        public static void Main()
        {
        // Introductorty
        
            Console.WriteLine("Calculator for roots of the general quadratic equation.");
            Console.WriteLine("Quadratic equation: ax^2 + bx + c = 0");
            Console.WriteLine();
            
            //Variable declearation

            double a, b, c, xOne, xTwo, midSqrt, resultOne, resultTwo;
            
            // Taking input for a, b and c
            
            Console.Write("Write the value of a: ");
            a = double.Parse(Console.ReadLine());
            Console.Write("Write the value of b: ");
            b = double.Parse(Console.ReadLine());
            Console.Write("Write the value of c: ");
            c = double.Parse(Console.ReadLine());

            midSqrt = (b * b) - (4 * a * c);
            midSqrt = Math.Sqrt(midSqrt);

            xOne = ((-(b) + midSqrt) / (2 * a));
            xTwo = ((-(b) - midSqrt) / (2 * a));

            resultOne = (a * xOne * xOne) + (b * xOne) + c;
            resultTwo = (a * xTwo * xTwo) + (b * xTwo) + c;

            Console.WriteLine();
            Console.WriteLine();
            Console.WriteLine("Result for the equation {0}x^2+{1}x+{2} is: ", a, b, c);
            Console.WriteLine("Root 1 x = {0}", Math.Round(xOne, 2));
            Console.WriteLine("Root 2 x = {0}", Math.Round(xTwo, 2));

            if (resultOne == 0)
            {
                PrintResult(xOne);
            }
            else if (resultTwo == 0)
            {
                PrintResult(xTwo);
            }
            else
            {
                PrintResult();
            }
            Console.ReadLine();
        }
        public static void PrintResult(double result)
        {
            Console.WriteLine("Valid root for the given quadratic equation is {0}",
                    Math.Round(result, 2));
        }
        public static void PrintResult()
        {
            Console.WriteLine("Both roots are valid for the given quadratic equation.");
        }

    }
}

File: RootOfQuadraticEquation.c
#include<stdio.h>
#include<math.h>
void main(){

    int a, b, c;
        printf("Write the value of a:");
    scanf("%d", &a);
        printf("Write the value of b:");
    scanf("%d", &b);
        printf("Write the value of c:");
    scanf("%d", &c);

    double xOne, xTwo, midSqrt, result;

    midSqrt = (b*b)-4*a*c;
    midSqrt = sqrt(midSqrt);

    xOne = (-b + midSqrt) / (2 * a);
    xTwo = (-b - midSqrt) / (2 * a);

    printf("%.2lf\n", midSqrt);
    printf("%.2lf\n", xOne);
    printf("%.2lf\n", xTwo);


    if((a*xOne*xOne+b*xOne+c) == 0 && (a*xTwo*xTwo+b*xTwo+c) == 0){
        printf("The valid value of x for the equation is: %.2lf and %.2lf",
        xOne, xTwo);
    }
    else if((a*xTwo*xTwo+b*xTwo+c) == 0)
    {
        printf("The valid value of x for the equation is: %.2lf", xTwo);
    }
    else{
        printf("The valid value of x for the equation is: %.2lf", xOne);
    }
}
